test/db/archos/linux-x64/dbg_oo

```diff
@@ -95,3 +95,13 @@ Hello world!
 EOF
 RUN
 
+NAME=ood check for ptrace errors
+FILE=bins/elf/analysis/x86-helloworld-gcc
+CMDS=<<EOF
+ood
+EOF
+REGEXP_ERR=([a-zA-Z-]+\s+)
```

Comment:
The implementation from OpenBSD in rz_regex is somewhat limited, I wasn't able to use negative/positive lookahead, non-capturing groups, \W, \D, \S, \b and had other weird behavior compared to python.

Reply:
Maybe it could be updated to the more recent version? I doubt modern version of it doesn't have support for those.

Reply:
I couldn't find anything related in [regcomp.c](https://github.com/openbsd/src/commits/master/lib/libc/regex). Based on their [commit history](https://github.com/openbsd/src/commits/master/lib/libc/regex) this code isn't really maintained.

Also, their [man page](https://man.openbsd.org/regex.3) states `This is an alpha release with known defects.` and `regexec() performance is poor. This will improve with later releases. nmatch exceeding 0 is expensive; nmatch exceeding 1 is worse.`(rz_regex_exec). I think that we should replace this implementation completely but we can still merge this PR with limited regex support for now if we are satisfied with the rz_regex API.

Reply:
I am fine with this API. Also, it wouldn't be too painful to replace the implementation with others. just need to make sure that the search functionalities will be okay still